extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -442,20 +451,136 @@ export abstract class AzureAuth implements vscode.Disposable {
 
 	//#region network functions
 
-	private async makeGetRequest<T>(url: string, token: string): Promise<AxiosResponse<T>> {
+	private async makeGetRequest<T>(requestUrl: string, token: string): Promise<AxiosResponse<T>> {
 		const config: AxiosRequestConfig = {
 			headers: {
 				'Content-Type': 'application/json',
-				'Authorization': `Bearer ${token}`
+				'Authorization': `Bearer ${token}`,
 			},
-			validateStatus: () => true // Never throw
+			validateStatus: () => true, // Never throw
+			// Turning off automatic proxy detection to avoid issues with tunneling agent by setting proxy to false.
+			// https://github.com/axios/axios/blob/bad6d8b97b52c0c15311c92dd596fc0bff122651/lib/adapters/http.js#L85
+			proxy: false
```

Comment:
so we're confident that we can turn this off, and the below new code will cover all the previous case that this setting did?  I'm not familiar with this functionality to access the risk of this change, honestly.

Reply:
I can change it, so that proxy false is only set when a proxy endpoint is found.